src/tabs/ConstantsPage.js

```diff
@@ -94,16 +92,15 @@ export const ConstantsPage = props => {
     };
 
     const clearButtonHandler = () => {
-        console.log("clearButtonHandler");
-        constantList.map((val,index) => {
-            constantList[index].value = 0; 
+        const clear_constants = constantList.map((item) => {
+            item.value = 0;
+            return item
         })
-
+        setConstantList(clear_constants) 
     }
```

Comment:
Então, de novo, vc tá usando map pra mudar valores (pq é um array de objetos, então vc acessa as coisas por referência), esse `item.value = 0` já tá alterando o valor original no constantList (e aí acaba fazendo isso duas vezes pq aloca um novo array e ainda usa o outro método pra copiar de novo)

Eu te juro que aquele `for..of` que eu comentei no outro é suficiente uashaush